store/cachekv/store.go

```diff
@@ -40,11 +24,8 @@ func NewStore(parent types.KVStore) *Store {
 
 // GStore wraps an in-memory cache around an underlying types.KVStore.
 type GStore[V any] struct {
-	mtx           sync.Mutex
-	cache         map[string]*cValue[V]
-	unsortedCache map[string]struct{}
-	sortedCache   btree.BTree[V] // always ascending sorted
-	parent        types.GKVStore[V]
+	writeSet btree.BTree[V] // always ascending sorted
```

Comment:
An important difference is we don't do cache anymore, only buffer the writes, because previously it was not efficient to do cache at all the layers, and in block-stm, the multiversion data structure will do cache automatically, so we need to revisit cache design later.